pyiron/atomistics/structure/_visualize.py

```diff
@@ -0,0 +1,609 @@
+            # Color by per-atom scalars
+            elif scalar_field is not None:  # Color by per-atom scalars
+                colors = _scalars_to_hex_colors(
+                    scalar_field, scalar_start, scalar_end, scalar_cmap
+                )
+                view = _add_custom_color_spacefill(
+                    view, atomic_numbers, particle_size, colors
+                )
+            # Color by element
+            else:
+                view = _add_colorscheme_spacefill(
+                    view, elements, atomic_numbers, particle_size
+                )
+            view.remove_ball_and_stick()
+        else:
+            view.add_ball_and_stick()
+
+        if show_cell:
+            if parent_basis.cell is not None:
+                if all(np.max(parent_basis.cell, axis=0) > 1e-2):
+                    view.add_unitcell()
+
+        if vector_color is None and vector_field is not None:
+            vector_color = (
+                0.5
+                * np.array(vector_field)
+                / np.linalg.norm(vector_field, axis=-1)[:, np.newaxis]
+                + 0.5
+            )
+        elif (
+            vector_field is not None and vector_field is not None
+        ):  # WARNING: There must be a bug here...
+            try:
+                if vector_color.shape != np.ones((len(self), 3)).shape:
+                    vector_color = np.outer(
+                        np.ones(len(self)), vector_color / np.linalg.norm(vector_color)
+                    )
+            except AttributeError:
+                vector_color = np.ones((len(self), 3)) * vector_color
+
+        if vector_field is not None:
+            for arr, pos, col in zip(vector_field, positions, vector_color):
+                view.shape.add_arrow(list(pos), list(pos + arr), list(col), 0.2)
+
+        if show_axes:  # Add axes
+            axes_origin = -np.ones(3)
+            arrow_radius = 0.1
+            text_size = 1
+            text_color = [0, 0, 0]
+            arrow_names = ["x", "y", "z"]
+
+            for n in [0, 1, 2]:
+                start = list(axes_origin)
+                shift = np.zeros(3)
+                shift[n] = 1
+                end = list(start + shift)
+                color = list(shift)
+                # We cast as list to avoid JSON warnings
+                view.shape.add_arrow(start, end, color, arrow_radius)
+                view.shape.add_text(end, text_color, text_size, arrow_names[n])
+
+        if camera != "perspective" and camera != "orthographic":
+            warnings.warn(
+                "Only perspective or orthographic is (likely to be) permitted for camera"
+            )
+
+        view.camera = camera
+        view.background = background
+
+        orientation = _get_flattened_orientation(view_plane=view_plane,
+                                                 distance_from_camera=distance_from_camera)
+        view.control.orient(orientation)
+
+        return view
+
+    def plot3d_ase(
```

Comment:
I don't quite understand what the difference between `plot3d()` and `plot3d_ase()` is, but since this was already this way previously, it can be a separate issue.  I'd want to discuss about this on Friday though.

Reply:
I can't really remember exactly why we had to create two functions, but I remember that for some time we were not so sure about our `plot3d`. Maybe that was the problem. To be honest, I don't think anyone still uses `plot3d_ase`, so maybe it's good to have it only in `plot3d(mode='ase')`.